libsql/src/hrana/transaction.rs

```diff
@@ -0,0 +1,62 @@
+use crate::hrana::connection::stmts_to_batch;
+use crate::hrana::pipeline::{ExecuteStreamReq, StreamRequest};
+use crate::hrana::proto::{BatchResult, Stmt, StmtResult};
+use crate::hrana::stream::HttpStream;
+use crate::hrana::{HttpSend, Result};
+use crate::TransactionBehavior;
+
+#[derive(Debug, Clone)]
+pub(crate) struct HttpTransaction<T>
+where
+    T: for<'a> HttpSend<'a>,
+{
+    stream: HttpStream<T>,
+}
+
+impl<T> HttpTransaction<T>
+where
+    T: for<'a> HttpSend<'a>,
+{
+    pub fn stream(&self) -> &HttpStream<T> {
+        &self.stream
+    }
+
+    pub async fn open(stream: HttpStream<T>, tx_behavior: TransactionBehavior) -> Result<Self> {
+        let begin_stmt = match tx_behavior {
+            TransactionBehavior::Deferred => "BEGIN DEFERRED",
+            TransactionBehavior::Immediate => "BEGIN IMMEDIATE",
+            TransactionBehavior::Exclusive => "BEGIN EXCLUSIVE",
+            TransactionBehavior::ReadOnly => "BEGIN READONLY",
```

Comment:
We probably need to map this to `DEFERRED` since I don't think the server supports this syntax. We should make sure we test this.

Reply:
I was suspecting that since replication client supports this (this code is copied from there), it should be supported in hrana as well.

Reply:
No clue, we should test that it actually works I have a feeling it might not but I could also be wrong.